scripts/openwebnet-discovery.nse

```diff
@@ -0,0 +1,199 @@
+local stdnse = require "stdnse"
+local shortport = require "shortport"
+local comm = require "comm"
+local string = require "string"
+
+description = [[
+OpenWebNet is a communications protocol developed by Bticino since 2000.
+Retrieves the Gateway and device type. Retrieves the count and addresses
+of lights, multimedia and many other services running on server/servers.
+]]
+
+---
+-- @usage
+-- nmap --script openwebnet-discovery
+--
+-- @output
+-- | openwebnet-discovery:
+-- |   Gateway: *#*1##*#13**15*12##*#1##
+-- |   Heating: 1
+-- |   Power Management: 1
+-- |   Multimedia: 1
+-- |   Device: F453AV
+-- |   Automation: 3
+-- |   Door Entry System: 2
+-- |   Burglar Alarm: 1
+-- |_  Lighting: 114
+--
+-- Version: 0.1, Updated on 21/06/2017
+
+author = "Rewanth Cool"
+license = "Same as Nmap--See https://nmap.org/book/man-legal.html"
+categories = {"discovery", "safe"}
+
+portrule = shortport.port_or_service(20000, "openwebnet")
+
+local device = {}
+device[2] = "MHServer"
```

Comment:
The preferred syntax for this is to allocate and create the table all in one statement. This is slightly more efficient since Lua can know the size of the table right away and doesn't have to keep growing and reallocating as we add elements:

```lua
local device = {
  [2] = "MHServer",
  [4] = "MH200",
  -- etc.
}
```